### PR TITLE
Tsprune

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Added ts-prune tool to dev dependencies in order to have a tool to report unused exported functions, var or constants.

Usage:

- In webapp folder: `npm run report-unused-exports`
- file `.ts-prunerc` with ignore patterns
- comment `// ts-prune-ignore-next` can be added before a line we want to keep even if it's unused

**current report**
```
➜  webapp git:(tsprune) npm run report-unused-exports 

> report-unused-exports
> ts-prune

src/actions.ts:169 - addNewTask
src/client.ts:181 - fetchPlaybookRunChannels
src/client.ts:276 - fetchUsersInChannel
src/client.ts:280 - fetchMyChannels
src/client.ts:284 - fetchUsersInTeam
src/client.ts:343 - clientRemoveChecklistItem
src/client.ts:490 - fetchStats
src/client.ts:533 - setGlobalSettings
src/client.ts:668 - clientFetchIsPlaybookFollower
src/client.ts:735 - favoriteItem
src/client.ts:762 - unfavoriteItem
src/client.ts:829 - doPatch
src/constants.ts:14 - TEMPLATE_TITLE_KEY
src/index.tsx:139 - default
src/components/checklist_item_input.tsx:139 - ChecklistItemButton
src/components/datetime_input.tsx:173 - default
src/hooks/index.ts:213 - clearLocks
src/hooks/index.ts:255 - useCanCreatePlaybooks
src/hooks/index.ts:272 - useCanRestrictPlaybookCreation
src/hooks/index.ts:290 - useExperimentalFeaturesEnabled
src/hooks/index.ts:335 - useThing
src/hooks/index.ts:414 - useDropdownPosition
src/hooks/index.ts:68 - FetchMetadata
src/hooks/index.ts:638 - usePrevious
src/hooks/index.ts:195 - useCategories
src/hooks/index.ts:26 - PlaybookCreateQueryParameters
src/hooks/index.ts:24 - useHasChannelPermission
src/hooks/index.ts:28 - useHasPlaybookPermissionById
src/hooks/index.ts:38 - PlaybookPermissionsParams
src/hooks/index.ts:6 - usePlaybookViewTelemetry
src/hooks/index.ts:14 - usePlaybookRunViewTelemetry
src/types/permissions.ts:6 - Permission
src/types/playbook.ts:97 - FetchPlaybooksCountReturn
src/components/assets/buttons.tsx:173 - UpgradeButton
src/components/assets/left_dots.tsx:182 - default
src/components/assets/left_fade.tsx:14 - default
src/components/assets/right_dots.tsx:638 - default
src/components/assets/right_fade.tsx:14 - default
src/components/backstage/new_playbook.tsx:31 - PlaybookModalButton
src/components/backstage/playbook_list.tsx:342 - useUpgradeModalVisibility
src/components/backstage/styles.tsx:25 - BackstageHeader
src/components/backstage/styles.tsx:36 - TeamContainer
src/components/backstage/styles.tsx:167 - StyledAsyncSelect
src/components/backstage/styles.tsx:227 - BackstageHorizontalContentSquish
src/components/backstage/styles.tsx:233 - RadioContainer
src/components/backstage/styles.tsx:240 - RadioLabel
src/components/modals/run_playbook_modal.tsx:240 - default
src/components/modals/update_run_status_modal.tsx:479 - default
src/components/rhs/rhs_checklists_add_dialog.tsx:57 - default
src/components/rhs/rhs_checklists_delete_dialog.tsx:72 - default
src/components/rhs/rhs_checklists_rename_dialog.tsx:66 - default
src/components/rhs/rhs_run_details_tour_dialog.tsx:146 - default
src/components/rhs/rhs_shared.tsx:21 - Footer
src/components/rhs/rhs_shared.tsx:61 - StyledFooterButton
src/components/rhs/rhs_shared.tsx:134 - SmallerProfile
src/components/rhs/rhs_timeline_placeholder.tsx:34 - default
src/components/team/team_selector.tsx:48 - default
src/components/tutorial/tours.ts:9 - AutoTourStatus
src/components/tutorial/tours.ts:55 - TTCategoriesMapToAutoTourStatusKey
src/components/assets/icons/alert_icon.tsx:21 - default
src/components/assets/icons/dot_menu_icon.tsx:12 - default
src/components/assets/icons/file_icon.tsx:10 - default
src/components/assets/icons/plus_icon.tsx:13 - default
src/components/backstage/playbook_editor/controls.tsx:133 - Share
src/components/backstage/playbook_editor/controls.tsx:154 - ArchivedLabel
src/components/backstage/playbook_runs/shared.tsx:67 - SecondaryButtonRight
src/components/backstage/playbook_runs/shared.tsx:84 - PrimaryButtonRight
src/components/backstage/runs_list/pagination_row.tsx:29 - default
src/components/backstage/runs_list/progress_bar.tsx:32 - default
src/components/backstage/runs_list/status_filter.tsx:26 - StatusFilter
src/components/tutorial/tutorial_tour_tip/index.ts:86 - useElementAvailable
src/components/backstage/playbook_runs/playbook_run/index.ts:6 - default
src/components/backstage/playbook_runs/playbook_run/rhs_info_styles.tsx:8 - Container
src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx:481 - default
src/components/backstage/playbooks/metrics/metrics_run_list.tsx:16 - SortParams
``` 

#### Ticket Link
No ticket

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/profile_vertical.tsx [hunk 1]
```diff
@@ -70,4 +70,5 @@ const ProfileVertical = (props: Props) => {
     );
 };
 
+// ts-prune-ignore-next
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/profile_vertical.tsx:73]: not sure about this one BTW, just added as example 😂

Comment (crspeller) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/profile_vertical.tsx:73]: Probably shouldn't add one we are not sure about. Usually when I see this I am more cautious.

**Review by crspeller — APPROVED** [review]

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/profile_vertical.tsx (hunk 1 above)
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/profile_vertical.tsx:73]:
```suggestion
```

**Comment by trilopin** [comment]
@calebroseland is anything that you'd want to change?

**Review by calebroseland — APPROVED** [review]
@trilopin, nope, this looks good to me 👍

**Pull request merged**